src/memefactory/server/twitter_bot.cljs

```diff
@@ -71,18 +71,22 @@
 
   (js/Promise.
    (fn [resolve reject]
-     (let [tmp-dir "/tmp/memefactory"
-           ;; NOTE: Maybe instead of a random id we can create a folder with the same name as the ipfs hash
-           ;; easier to debug
-           tmp-name (str (random-uuid))
-           tar-file (str tmp-dir "/" tmp-name ".tar")
-           img-tmp-dir (str tmp-dir "/" tmp-name)
-           extract-stream (.extract tar-fs img-tmp-dir)]
-       (.writeFileSync fs tar-file buffer)
-       (.pipe (.createReadStream fs tar-file) extract-stream)
-       (.on extract-stream "finish"
-            (fn []
-              (resolve (.readFileSync fs (str img-tmp-dir "/" (aget (.readdirSync fs img-tmp-dir) 0))))))))))
+
+     (try
+       (let [tmp-dir "/tmp/memefactory"
+             ;; NOTE: Maybe instead of a random id we can create a folder with the same name as the ipfs hash
+             ;; easier to debug
+             tmp-name (str (random-uuid))
+             tar-file (str tmp-dir "/" tmp-name ".tar")
+             img-tmp-dir (str tmp-dir "/" tmp-name)
+             extract-stream (.extract tar-fs img-tmp-dir)]
+         (.writeFileSync fs tar-file buffer)
+         (.pipe (.createReadStream fs tar-file) extract-stream)
+         (.on extract-stream "finish"
+              (fn []
+                (resolve (.readFileSync fs (str img-tmp-dir "/" (aget (.readdirSync fs img-tmp-dir) 0)))))))
+       (catch js/Error e
```

Comment:
catch :default e

Reply:
What is the difference?

Reply:
JS allows to throw everything (also js/Object), this will catch everything